File: projekt3/Antelope.py
from Animal import *

class Antelope(Animal):
    __ANTELOPE_STRENGTH__ = 4
    __ANTELOPE_INITIATIVE__ = 4
    __ANTELOPE_MOVE_RANGE__ = 2
    __COLOR__ = "Brown"
    
    def __init__(self, row, col, world):
        super().__init__(row, col, self.__ANTELOPE_STRENGTH__, self.__ANTELOPE_INITIATIVE__, world)
    
    def getName(self):
        return "Antelope"
    
    def getColor(self):
        return self.__COLOR__   
    
    def reproduce(self, row, col):
        self._world_.setOrganism(row, col, Antelope())

    def specialAbility(self, other):
        rand = randrange(2)
        if (rand == 1):
            if (self._world_.getOrgFromPos(self.getRow() - 1, self.getCol()) == None):
                self.move(self.getRow() - 1, self.getCol())
            elif (self._world_.getOrgFromPos(self.getRow(), self.getCol() + 1) == None):
                self.move(self.getRow(), self.getCol() + 1)
            elif (self._world_.getOrgFromPos(self.getRow() + 1, self.getCol()) == None):
                self.move(self.getRow() + 1, self.getCol())
            elif (self._world_.getOrgFromPos(self.getRow(), self.getCol() - 1) == None):
                self.move(self.getRow(), self.getCol() - 1)
            event = self.getName() + "uciekla przed" + other.getName()
            self._world_.addEvent(event)
            return True
        else:
            self._world_.removeOrganism(self.getRow(), self.getCol())
            self.die()
        return True

    def takeAction(self):
        rand = randrange(4)
        if rand == 0: #gore
            self.move(self.getRow()-self.__ANTELOPE_MOVE_RANGE__ , self.getCol())
        elif rand == 1: #prawo
            self.move(self.getRow() , self.getCol()+self.__ANTELOPE_MOVE_RANGE__)
        elif rand == 2: #dol
            self.move(self.getRow()+self.__ANTELOPE_MOVE_RANGE__ , self.getCol())
        elif rand == 3: # lewo
            self.move(self.getRow() , self.getCol()-self.__ANTELOPE_MOVE_RANGE__)
        else:
            pass

File: projekt2/src/proj/java/model/animal/Antelope.java
package proj.java.model.animal;

import java.awt.Color;
import java.util.Random;

import proj.java.model.Animal;
import proj.java.model.Organism;
import proj.java.model.World;

public class Antelope extends Animal {

    private static final int ANTELOPE_STRENGTH = 4;
    private static final int ANTELOPE_INITIATIVE = 4;
    private static final int ANTELOPE_MOVE_RANGE = 2;

    @Override
    public Color getColor() {
        return Color.MAGENTA;
    }

    @Override
    public int getInitiative() {
        return ANTELOPE_INITIATIVE;
    }

    @Override
    public int getStrength() {
        return ANTELOPE_STRENGTH + getExtraStrength();
    }

    @Override
    protected void reproduce(World world, int row, int col) {
        world.setOrganism(row, col, new Antelope());
    }

    @Override
    public boolean specialAbility(Organism other, World world, boolean specialAblityStopsMovement) {
        var rand = new Random().nextInt(2);
        if (rand == 1) {
            if (world.getOrganism(getRow() - 1, getCol()) == null)
                move(getRow(), getCol(), getRow() - 1, getCol(), world);
            else if (world.getOrganism(getRow(), getCol() + 1) == null)
                move(getRow(), getCol(), getRow(), getCol() + 1, world);
            else if (world.getOrganism(getRow() + 1, getCol()) == null)
                move(getRow(), getCol(), getRow() + 1, getCol(), world);
            else if (world.getOrganism(getRow(), getCol() - 1) == null)
                move(getRow(), getCol(), getRow(), getCol() - 1, world);
            String message = String.format("%s uciekla przed %s", this.getName(), other.getName());
            world.addMessage(message);
            specialAblityStopsMovement = false;
            return true;
        }
        specialAblityStopsMovement = false;
        world.removeOrganism(this.getRow(), this.getCol());
        return true;
    }

    @Override
    public void takeAction(World world) {
        var rand = new Random().nextInt(4);
        switch (rand) {
            case 0: // gora
                move(getRow(), getCol(), getRow() - ANTELOPE_MOVE_RANGE, getCol(), world);
                break;
            case 1: // prawo
                move(getRow(), getCol(), getRow(), getCol() + ANTELOPE_MOVE_RANGE, world);
                break;
            case 2: // lewo
                move(getRow(), getCol(), getRow() + ANTELOPE_MOVE_RANGE, getCol(), world);
                break;
            case 3: // dol
                move(getRow(), getCol(), getRow(), getCol() - ANTELOPE_MOVE_RANGE, world);
                break;
            default:
                break;
        }
    }

}
